python/activity_stream/reply_list.py

```diff
@@ -0,0 +1,415 @@
+            
+            for attachment_req in attachment_requests:
+                self._data_manager.request_attachment_thumbnail(-1, 
+                                                                attachment_req["attachment_group_id"], 
+                                                                attachment_req["attachment_data"])
+            
+            # now go through the shotgun data
+            # for each reply, request a thumbnail.
+            requested_items = []
+            for item in replies_and_attachments:
+                if item["type"] == "Reply":
+                    # note that the reply data structure is special:
+                    # the 'user' key is not a normal sg link dict, 
+                    # but contains an additional image field to describe
+                    # the thumbnail:
+                    # 
+                    # {'content': 'Reply content...', 
+                    #  'created_at': 1438649419.0, 
+                    #  'type': 'Reply', 
+                    #  'id': 73, 
+                    #  'user': {'image': '...', 
+                    #           'type': 'HumanUser', 
+                    #           'id': 38, 
+                    #           'name': 'Manne Ohrstrom'}}]
+                    reply_author = item["user"]
+                    
+                    uniqueness_key = (reply_author["type"], reply_author["id"])
+                    if uniqueness_key not in requested_items:
+                        # this thumbnail has not been requested yet    
+                        if reply_author.get("image"):
+                            # there is a thumbnail for this user!
+                            requested_items.append(uniqueness_key)
+                            self._data_manager.request_user_thumbnail(reply_author["type"], 
+                                                                      reply_author["id"],
+                                                                      reply_author["image"])
+            
+        finally:
+            # make the window visible again and trigger a redraw
+            self.setVisible(True)
+            
+        self._bundle.log_debug("...done")
+
+        
+    def _clear(self):
+        """
+        Clear the widget. This will remove all items from the UI
+        """
+        self._bundle.log_debug("Clearing UI...")
+        
+        for x in self._general_widgets + self._reply_widgets + self._attachment_group_widgets.values():
+            # remove widget from layout:
+            self.ui.reply_layout.removeWidget(x)
+            # set it's parent to None so that it is removed from the widget hierarchy
+            x.setParent(None)
+            # mark it to be deleted when event processing returns to the main loop
+            x.deleteLater()
+    
+        self._general_widgets = []
+        self._reply_widgets = []
+        self._attachment_group_widgets = {}    
+        
+    def _load_stylesheet(self):
```

Comment:
I've seen this method a few places. Looks like it could be a utility method somewhere.